tests/page-objects/apdex/login.page.js

```diff
@@ -0,0 +1,29 @@
+const Page = require('@page-objects/apdex/page');
+
+class LoginPage extends Page {
+  
+  get inputUsername() {
+    return $('//android.view.View[@text="User name"]//parent::android.view.View/android.widget.EditText');
+  }
+
+  get inputPassword() {
+    return $('//*[@text="Password"]//parent::android.view.View/android.view.View/android.widget.EditText');
+  }
+
+  async login(userType, userRole, settingsProvider) {
```

Comment:
Nope it's not a miss - for some reason it seems because the class has not been initialized, on line 16 when we call `const user = super.settingsProvider.getUser(userType, userRole);` it returns undefined

Reply:
Then we should probably make sure it gets initialized?

Reply:
@dianabarsan can you kindly provide an example of how to do this before calling the class?

Reply:
ok i got this 👍 thanks